File: app/src/main/java/com/szollosi/sensormusicplayer/fragments/FragmentZ.java
package com.szollosi.sensormusicplayer.fragments;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.github.mikephil.charting.charts.LineChart;
import com.github.mikephil.charting.data.LineData;
import com.szollosi.sensormusicplayer.Constants;
import com.szollosi.sensormusicplayer.R;

public class FragmentZ extends BaseFragment {

  private Handler mHandler;

  private LineChart chart;

  private Runnable refreshGraph = new Runnable() {

    @Override
    public void run() {
      synchronized (dataPointsZ) {
        seriesY.setValues(dataPointsZ);
        chart.setData(new LineData(seriesZ));
        chart.invalidate();
        mHandler.postDelayed(this, Constants.DELAY_REFRESH);
      }
    }
  };

  private int page;

  private String title;

  public static FragmentZ newInstance(int page, String title) {
    FragmentZ fragmentFirst = new FragmentZ();
    Bundle args = new Bundle();
    args.putInt("someInt", page);
    args.putString("someTitle", title);
    fragmentFirst.setArguments(args);
    return fragmentFirst;
  }

  @Nullable
  @Override
  public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
      @Nullable Bundle savedInstanceState) {
    View rootView = inflater.inflate(R.layout.fragment_section_z, container, false);
    chart = rootView.findViewById(R.id.graph_z);
    return rootView;
  }

  @Override
  public void onResume() {
    super.onResume();
    mHandler = new Handler(Looper.myLooper());
    mHandler.postDelayed(refreshGraph, Constants.DELAY_REFRESH);
  }

  @Override
  public void onPause() {
    super.onPause();
    mHandler.removeCallbacks(refreshGraph);
  }

  @Override
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    page = getArguments().getInt("someInt", 0);
    title = getArguments().getString("someTitle");
  }
}

File: app/src/main/java/com/szollosi/sensormusicplayer/fragments/FragmentZ.kt
package com.szollosi.sensormusicplayer.fragments

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.github.mikephil.charting.charts.LineChart
import com.github.mikephil.charting.data.LineData
import com.szollosi.sensormusicplayer.Constants
import com.szollosi.sensormusicplayer.R

class FragmentZ : BaseFragment() {

  private var mHandler: Handler? = null

  private var chart: LineChart? = null

  private val refreshGraph = object : Runnable {

    override fun run() {
      synchronized(dataPointsZ) {
        seriesY.values = dataPointsZ.toArrayList()
        chart!!.data = LineData(seriesZ)
        chart!!.invalidate()
        mHandler!!.postDelayed(this, Constants.DELAY_REFRESH.toLong())
      }
    }
  }

  private var page: Int = 0

  private var title: String? = null

  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                            savedInstanceState: Bundle?): View? {
    val rootView = inflater.inflate(R.layout.fragment_section_z, container, false)
    chart = rootView.findViewById(R.id.graph_z)
    return rootView
  }

  override fun onResume() {
    super.onResume()
    mHandler = Handler(Looper.myLooper()!!)
    mHandler!!.postDelayed(refreshGraph, Constants.DELAY_REFRESH.toLong())
  }

  override fun onPause() {
    super.onPause()
    mHandler!!.removeCallbacks(refreshGraph)
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    page = arguments!!.getInt("someInt", 0)
    title = arguments!!.getString("someTitle")
  }

  companion object {

    fun newInstance(page: Int, title: String): FragmentZ {
      val fragmentFirst = FragmentZ()
      val args = Bundle()
      args.putInt("someInt", page)
      args.putString("someTitle", title)
      fragmentFirst.arguments = args
      return fragmentFirst
    }
  }
}
